libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -230,6 +236,18 @@ class MapboxRouteLineApi(
         )
     }
 
+    /**
+     * Updates the [RouteLineColorResources] used to determine the colors of the route line.
+     *
+     * @param ctx a context
+     * @param res a [RouteLineColorResources]
+     */
+    fun updateColorResources(ctx: Context, res: RouteLineColorResources) {
```

Comment:
If we're allowing for color mutations, why not the whole `MapboxRouteLineOptions`? Is there any performance advantage in doing this versus re-instantiating the class? I think that we either need to commit to the mutability of this class (and understand why), or leave things as they were.

This function should also return a result that can be applied to the view immediately.

Reply:
Should we also take that the opportunity to deprecate `MapboxRouteLineOptions` usage in `MapboxRouteLineView` and introduce a separate `MapboxRouteLineViewOptions` to make the usage less ambiguous and confusing?

Reply:
One reason is the vanishing route line instance is in the options.  If the class starts with options that have a vanishing route line instance and then call update with a null vanishing route line and/or call update with options that have a different vanishing route line instance it creates a weird state.  I haven't done a deep dive into testing it but since it was only color updates that were asked for I think trying to manage what could happen with the vanishing route line instance is over complicated.

Reply:
>Should we also take that the opportunity to deprecate MapboxRouteLineOptions usage in MapboxRouteLineView and introduce a separate MapboxRouteLineViewOptions to make the usage less ambiguous and confusing?

Perhaps. Having the same options in both classes has provided the flexibility to make the changes we've made in the implementations over the past months. It depends on if we want to keep that flexibility or not.

Reply:
>Is there any performance advantage in doing this versus re-instantiating the class?

We've gotten feedback from a couple customers. I can share the slack threads with you if you wish.  Even though color changes can be done via the `MapboxRouteLineApiExtensions::setPrimaryTrafficColor` and similar methods, updating of the color options has been asked for.  I don't have a strong opinion either way. One of the reasons this has been a draft PR was to elicit a second opinion.  I don't think this feature has been promised to anybody. I only responded to a customer request by saying I'd investigate.

Reply:
>This function should also return a result that can be applied to the view immediately.

I don't agree. This is a setter function.  Updating the colors and rendering a line don't necessarily happen at the same place/time in a client application.  The client application can call `getRouteDrawData()` to get a renderable object anytime after updating the color resources.